### ISSUE TITLE
ClassCastException: click listener on EpoxyModel within a EpoxyModelGroup

### ISSUE THREAD

**Issue by eboudrant** [airbnb/epoxy#687]
I am not sure it is a new or old issue but when we set a click listener on an `EpoxyModel` which is part of an `EpoxyModelGroup` we receive `ClassCastException` on click : 

```
java.lang.ClassCastException: x.x.ModelGroup cannot be cast to x.x.NestedModel
 	at x.x.EpoxyController$buildModels$$inlined$apply$lambda$1.onClick(EpoxyController.kt:274)
 	at com.airbnb.epoxy.WrappedEpoxyModelClickListener.onClick(WrappedEpoxyModelClickListener.java:52)
 	at android.view.View.performClick(View.java:6597)
 	at android.view.View.performClickInternal(View.java:6574)
        at ...
```

```
val nestedModel = ModelGroup_()
    .id(...)
    .imageClickListener { model, _, clickedView, _ ->
        ...
    }

// then nestedModel get added to an EpoxyModelGroup
```

**Comment by elihart** [comment]
Thanks for the report, I think this has existed for a long time. It will only happen if you use the model click listener.

The code in WrappedEpoxyModelClickListener does
```

**Comment by elihart** [comment]
Potentially the generated code needs to change the `WrappedEpoxyModelClickListener` constructor to also take the model it is associated with, then when we find a model group we can check whether the listener's model is the top level group, or any nested models (potentially recursively)

**Comment by TheLester** [comment]
Any update on this?

**Comment by elihart** [comment]
No, but I'd to see somebody put up a PR.

It may be as simple as modifying `WrappedEpoxyModelClickListener` to check if the top level view holder model is a group and if so pulling out the right child model

**Comment by roncbird** [comment]
We've started using MvRx and Epoxy for a new app we're building here at work, and the libraries are great, but I just recently received an feature request that requires a fairly complicated UI design for rows, which has required me to use an EpoxyModelGroup. We are using databindings so I've added the clickListener in the xml. This has caused me to run into this same error as well, so I wanted to check and see if there is a fix in the works for this, or should I avoid using the model onClickListener?
